src/persist/src/indexed/mod.rs

```diff
@@ -689,24 +726,78 @@ impl<L: Log, B: Blob> Indexed<L, B> {
             self.blob.delete_unsealed_batch(&batch)?;
         }
 
-        for batch in deleted_trace_batches {
-            self.blob.delete_trace_batch(&batch)?;
+        Ok(())
+    }
+
+    fn compact_trace_external_reqs(&mut self) -> Result<Vec<ExternalWorkReq>, Error> {
+        let mut reqs = vec![];
+        for (id, arrangement) in self.state.arrangements.iter() {
+            if self.in_flight_trace_compactions.contains(&id) {
+                continue;
+            }
+            if let Some(req) = arrangement.trace_next_compact_req(&self.maintainer)? {
+                reqs.push(ExternalWorkReq::CompactTrace(req));
+                self.in_flight_trace_compactions.insert(*id);
+            }
         }
+        Ok(reqs)
+    }
+
+    fn do_handle_external_work_res(&mut self, res: ExternalWorkRes) -> Result<(), Error> {
+        match res {
+            ExternalWorkRes::CompactTrace(res) => {
+                match res {
+                    Ok(res) => {
+                        let written_bytes = res.merged.size_bytes;
+                        self.metrics.compaction_write_bytes.inc_by(written_bytes);
+                        self.apply_batched_cmd(|state, pending| {
+                            let id = res.req.id;
+                            if !state.arrangements.contains_key(&id) {
+                                log::trace!(
```

Comment:
return an error here. meant to do so, got lost in the code movement.

Reply:
yup, this should be a no-op. changed the code a bit but now this state just triggers a write to log and exit.

erroring a batch with a seal actually isn't that bad for a persisted source because the seal operator retries failed seals (unless there is another seal also pending at a time greater than the failed seal). A failed write permanently wedges the source :/